components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -25,6 +25,68 @@ const getWindowId = (id: number) => {
   return `id_${id}`
 }
 
+function pollTwitchPage (tab: chrome.tabs.Tab, tabId: number, publisherBlob:string): void {
+  let url = tab.url
+  // use an interval here to monitor when the DOM has finished
+  // generating. clear after the data is present.
+  // Check every second no more than 'limit' times
+  // clear the interval if panel closes
+  const markupMatch = '<figure class=\"tw-avatar tw-avatar--size-36\">' +
+                      '<div class=\"tw-border-radius-medium tw-overflow-hidden\">' +
+                      '<img class=\"tw-avatar__img tw-image\" alt=\"'
+  let obtained = false
+  // run once to see if we already have it
+  chrome.tabs.executeScript(tabId, {
+    code: 'var outerHtml = document.body.outerHTML; outerHtml'
+  }, function (result: string[]) {
+    if (result[0].includes(markupMatch)) {
+      obtained = true
+      publisherBlob = result[0]
+      if (url) {
+        chrome.braveRewards.onTwitchInfo(publisherBlob, url)
+      }
+      if (tab && tab.windowId && tab.url && tab.favIconUrl) {
+        chrome.braveRewards.getTwitchPublisherData(tab.windowId, tab.url, tab.favIconUrl || '', publisherBlob)
+      }
+    }
+  })
+
+  // if not obtained, then poll page for reasonable loading time
+  if (!obtained) {
+    let itr = 0
+    const limit = 10
+
+    let interval = setInterval(function () {
+      chrome.tabs.executeScript(tabId, {
+        code: 'var outerHtml = document.body.outerHTML; outerHtml'
```

Comment:
[minor] would just `code: 'document.body.outerHTML'` work here?

Reply:
I believe it should and meant to make it that way. According to extensions docs the last evaluation is returned. I'll make the change.

Reply:
Fixed.